modules/terminate-agent-hook/iam.tf

```diff
@@ -33,14 +33,14 @@ resource "aws_iam_role" "lambda" {
 }
 
 resource "aws_iam_role_policy_attachment" "lambda_kms" {
-  count = var.kms_key_id != "" ? 1 : 0
+  count = !var.enable_managed_kms_key ? 1 : 0
```

Comment:
question: if the root module has `enable_managed_kms_key == false`,  the `kms_key_id` was used in the policy. Now the policy attachment is no longer created and the Lambda function is not able to access the key. Anything I overlook?

Reply:
Hey Kayman,

yes that's correct, in fact here if we use a string comparison on a variable emptiness, terraform is not able to run the plan which is causing the issue. so the idea was to replace  `count = var.kms_key_id != "" ? 1 : 0` by `count = !var.enable_managed_kms_key ? 1 : 0` consistently.

The variable `enable_managed_kms_key` is now also transmitted from the parent module to the child module to have the count logic working.

Thanks a lot.